game-core/src/main/java/games/strategy/triplea/ai/tree/BattleTreeCalculator.java

```diff
@@ -0,0 +1,124 @@
+      final int runCount) {
+    final long startTime = System.currentTimeMillis();
+
+    // remove all of the non combatants
+    final MustFightBattle battle = new MustFightBattle(location, attacker, data, null);
+    final List<Unit> attackingUnitsCleaned = battle.removeNonCombatants(attackingUnits, true, true);
+    final List<Unit> defendingUnitsCleaned =
+        battle.removeNonCombatants(defendingUnits, false, true);
+
+    final boolean amphibious = false;
+    final List<Unit> amphibiousLandAttackers = null;
+
+    final List<Unit> attackingOrderOfLoss =
+        CasualtyOrderOfLosses.sortUnitsForCasualtiesWithSupport(
+            CasualtyOrderOfLosses.Parameters.builder()
+                .targetsToPickFrom(attackingUnitsCleaned)
+                .player(attacker)
+                .enemyUnits(defendingUnitsCleaned)
+                .combatModifiers(
+                    UnitBattleComparator.CombatModifiers.builder()
+                        .territoryEffects(territoryEffects)
+                        .amphibious(amphibious)
+                        .defending(false)
+                        .build())
+                .amphibiousLandAttackers(
+                    amphibiousLandAttackers == null ? List.of() : amphibiousLandAttackers)
+                .battlesite(location)
+                .costs(TuvUtils.getCostsForTuv(attacker, data))
+                .data(data)
+                .build());
+
+    final List<Unit> defendingOrderOfLoss =
+        CasualtyOrderOfLosses.sortUnitsForCasualtiesWithSupport(
+            CasualtyOrderOfLosses.Parameters.builder()
+                .targetsToPickFrom(defendingUnitsCleaned)
+                .player(defender)
+                .enemyUnits(attackingUnitsCleaned)
+                .combatModifiers(
+                    UnitBattleComparator.CombatModifiers.builder()
+                        .territoryEffects(territoryEffects)
+                        .amphibious(amphibious)
+                        .defending(true)
+                        .build())
+                .amphibiousLandAttackers(
+                    amphibiousLandAttackers == null ? List.of() : amphibiousLandAttackers)
+                .battlesite(location)
+                .costs(TuvUtils.getCostsForTuv(defender, data))
+                .data(data)
+                .build());
+
+    if (Properties.getTransportCasualtiesRestricted(data)) {
+      // move all transports to the end so that they are picked last
+      attackingOrderOfLoss.sort(
+          (unit1, unit2) ->
+              Boolean.compare(
+                  Matches.unitIsTransport().and(Matches.unitIsSea()).test(unit1),
+                  Matches.unitIsTransport().and(Matches.unitIsSea()).test(unit2)));
+      defendingOrderOfLoss.sort(
+          (unit1, unit2) ->
+              Boolean.compare(
+                  Matches.unitIsTransport().and(Matches.unitIsSea()).test(unit1),
+                  Matches.unitIsTransport().and(Matches.unitIsSea()).test(unit2)));
```

Comment:
`Comparator.comparing(Matches.unitIsTransport().and(Matches.unitIsSea())::test)`
might even work already